helper/sensitive/sensitive.go

```diff
@@ -0,0 +1,20 @@
+package sensitive
+
+import "encoding/json"
+
+type Sensitive string
+
+func (s Sensitive) Plaintext() string {
+	return string(s)
+}
+
+func (s Sensitive) String() string {
+	if s == "" {
+		return ""
+	}
+	return "[REDACTED]"
+}
+
+func (s Sensitive) MarshalJSON() ([]byte, error) {
```

Comment:
Should we also implement `UnmarshalJSON([]byte) error`? This way you can accept a plaintext value as a `Sensitive` type in an api?

Reply:
We don't need to - the default implementation already does so and behavior is tested in https://github.com/hashicorp/nomad/blob/50adb7c9f0d25d1aac3b7f0f1500d1ee2fe9b405/helper/sensitive/sensitive_test.go#L37-L47 .